backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java

```diff
@@ -214,18 +214,16 @@ public void close() {
         pool.close();
     }
 
-    public boolean tryChannel() {
-        Channel channel = null;
-        try {
-            channel = borrow().block();
-            return channel.isOpen();
-        } catch (Throwable t) {
-            return false;
-        } finally {
-            if (channel != null) {
-                borrowedChannels.remove(channel);
-                pool.returnObject(channel);
-            }
-        }
+    public Mono<Boolean> tryChannel() {
+        return Mono.usingWhen(borrow(),
+            channel -> Mono.just(channel.isOpen()),
+            channel -> {
```

Comment:
why a null check here and not on `isOpen()`?

Reply:
It's a mono wrapping as the managed resource is an Publisher, it's required for typing.